tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/AWS/DataStreamsMonitoringAwsSqsTests.cs

```diff
@@ -0,0 +1,97 @@
+// <copyright file="DataStreamsMonitoringAwsSqsTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading.Tasks;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.TestHelpers;
+using Datadog.Trace.TestHelpers.DataStreamsMonitoring;
+using FluentAssertions;
+using VerifyXunit;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Datadog.Trace.ClrProfiler.IntegrationTests.AWS;
+
+[Collection(nameof(AwsSqsTests.AwsSqsTestsCollection))]
+[Trait("RequiresDockerDependency", "true")]
+[UsesVerify]
+public class DataStreamsMonitoringAwsSqsTests : TestHelper
+{
+    public DataStreamsMonitoringAwsSqsTests(ITestOutputHelper output)
+        : base("AWS.SQS", output)
+    {
+    }
+
+    public static IEnumerable<object[]> GetEnabledConfig()
+    {
+        return from packageVersionArray in PackageVersions.AwsSqs
+               select new[] { packageVersionArray[0] };
+    }
+
+    [SkippableTheory]
+    [MemberData(nameof(GetEnabledConfig))]
+    [Trait("Category", "EndToEnd")]
+    public async Task SubmitsDsmMetrics(string packageVersion)
+    {
+        SetEnvironmentVariable(ConfigurationKeys.DataStreamsMonitoring.Enabled, "1");
+
+        using var telemetry = this.ConfigureTelemetry();
+        using var agent = EnvironmentHelper.GetMockAgent();
+        using (RunSampleAndWaitForExit(agent, packageVersion: packageVersion))
+        {
+#if NETFRAMEWORK
+            var expectedCount = 56;
+            var frameworkName = "NetFramework";
```

Comment:
Yeah, there's currently a gap for a bunch of our integrations that use .NET FX and Docker. I wouldn't worry too much tbh, you could even just throw a `SkipException()` here if you like, to make it clear that part is untested 🙂

Reply:
turns out this doesn't work, because even if the tests are not run, they are still compiled, and the compiler frowns if I put a `throw` here because it flags the code that follows as unreachable.